### PR TITLE
feat: cmd preview and apply support new flag '--ignore-fields'

### PR DESCRIPTION
**Pull request by howieyuen**
close: #126 

- feat: cmd preview and apply support new flag '--ignore-fields'
- refactor: distinguish compile flags and options
- refactor: move preview logic from `cmd/apply` to `cmd/preview`
- refactor: delete `io.writer` from `func Preview()` and add to `func Summary()`

### CODE REVIEW

File: pkg/engine/operation/graph/resource_node.go [hunk 1]
```diff
@@ -83,6 +85,12 @@ func (rn *ResourceNode) Execute(operation *opsmodels.Operation) status.Status {
 				return dryRunResp.Status
 			}
 			predictableState = dryRunResp.Resource
+			// Ignore differences of target fields
```
Comment (SparkYuan) [inline pkg/engine/operation/graph/resource_node.go:88]: test coverage -1.05%, please add some uts

File: pkg/kusionctl/cmd/preview/options.go [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
-	Detail  bool
-	NoStyle bool
+	PreviewFlags
+	backend.BackendOps
+}
+
+type PreviewFlags struct {
+	Operator     string
+	Detail       bool
+	IgnoreFields []string
```
Comment (SparkYuan) [inline pkg/kusionctl/cmd/preview/options.go:33]: add usages in our doc for this feature -ignore-fields

Comment (howieyuen) [reply pkg/kusionctl/cmd/preview/options.go:33]: I added usage in the cmd example

File: pkg/engine/operation/graph/resource_node.go (hunk 1 above)
Comment (howieyuen) [reply pkg/engine/operation/graph/resource_node.go:88]: ok

**Comment by SparkYuan** [comment]
Any instructions about this feature？

**Comment by howieyuen** [comment]
> Any instructions about this feature？

execute `kusion preview -h`, there is a show case, like below:
```
# Preview with ignored fields
kusion preview --ignore-fields="metadata.generation,metadata.managedFields"
```

**Review by SparkYuan — APPROVED** [review]
lgtm

**Review by markliby — APPROVED** [review]
lgtm

**Pull request merged**